controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,414 @@
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-releasestrategy-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseStrategySpec{
+				Pipeline:              "release-pipeline",
+				Bundle:                "test-bundle",
+				Policy:                "test-policy",
+				PersistentVolumeClaim: "test-pvc",
+				ServiceAccount:        "test-account",
+			},
+		}
+		Expect(noCacheClient.Create(ctx, releaseStrategy)).Should(Succeed())
+
+		releaseLink = &appstudiov1alpha1.ReleaseLink{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "ReleaseLink",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-releaselink-",
+				Namespace:    testNamespace,
+				Labels: map[string]string{
+					"release.appstudio.openshift.io/auto-release": "false",
+				},
+			},
+			Spec: appstudiov1alpha1.ReleaseLinkSpec{
+				DisplayName:     "Test ReleaseLink",
+				Application:     "test-app",
+				Target:          "default",
+				ReleaseStrategy: releaseStrategy.GetName(),
+			},
+		}
+		Expect(noCacheClient.Create(ctx, releaseLink)).Should(Succeed())
+
+		applicationSnapshot = &appstudioshared.ApplicationSnapshot{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-snapshot-",
+				Namespace:    "default",
+			},
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: testApiVersion,
+				Kind:       "ApplicationSnapshot",
+			},
+			Spec: appstudioshared.ApplicationSnapshotSpec{
+				Application: "testapplication",
+				DisplayName: "Test application",
+				Components:  []appstudioshared.ApplicationSnapshotComponent{},
+			},
+		}
+		Expect(noCacheClient.Create(ctx, applicationSnapshot)).Should(Succeed())
+
+		// adding the Type Metas as it loses it after creation
+		// and these fields are required by some functions
+		applicationSnapshot.TypeMeta = metav1.TypeMeta{
+			Kind:       "ApplicationSnapshot",
+			APIVersion: testApiVersion,
+		}
+		Expect(noCacheClient.Update(ctx, applicationSnapshot)).Should(Succeed())
+
+		release = &appstudiov1alpha1.Release{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: testApiVersion,
+				Kind:       "Release",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-release-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseSpec{
+				ApplicationSnapshot: applicationSnapshot.GetName(),
+				ReleaseLink:         releaseLink.GetName(),
+			},
+		}
+		Expect(noCacheClient.Create(ctx, release)).Should(Succeed())
+
+		adapter = NewAdapter(release, ctrl.Log, noCacheClient, ctx)
+		Expect(reflect.TypeOf(adapter)).
+			To(Equal(reflect.TypeOf(&Adapter{})))
+
+		// some functions needs the cache to be up to work
+		cacheAdapter = NewAdapter(release, ctrl.Log, cacheClient, ctx)
+		Expect(reflect.TypeOf(cacheAdapter)).
+			To(Equal(reflect.TypeOf(&Adapter{})))
+
+	})
+	AfterEach(func() {
+		ctx := context.Background()
+		noCacheClient.Delete(ctx, releaseLink)
```

Comment:
That is because In some tests I am deleting the release or other of these objects during the tests then the `AfterEach` breaks the full suite.

Do you know a way to avoid that behavior only for some specific cases?

Reply:
Completely outdated.